### PR TITLE
Add Data node to test unit JSON

### PR DESCRIPTION
**Pull request by unknwon**
Addresses #71, main lib are now detected and linked.

### CODE REVIEW

File: grapher/scan.py [hunk 1]
```diff
@@ -132,11 +132,22 @@ def pkgToUnit(pkg: Dict) -> List[Unit]:
     if len(test_files) == 0:
         return [unit]
 
+    # Collect files as candidates of modules.
+    modules = []
```
Comment (beyang) [inline grapher/scan.py]: Factor this out into a function called `unitToPkg`. It should do the reverse of `pkgToUnit`.

Comment (beyang) [inline grapher/scan.py]: Why is this necessary? Isn't this already available in `pkg.modules`?

File: grapher/scan.py [hunk 2]
```diff
@@ -117,7 +117,7 @@ def pkgToUnit(pkg: Dict) -> List[Unit]:
             deps.append(dep)
 
     unit = Unit(
```
Comment (beyang) [inline grapher/scan.py]: `pkgToUnit` should be renamed to `pkgToUnits`, since it can return more than 1 unit now.

File: grapher/scan.py (hunk 1 above)
Comment (Unknwon) [inline grapher/scan.py]: `setup.py` file only define ones the library wants to expose, but test files are using ones that are not defined in the `setup.py`'s `pkg['modules'] and pkg['py_modules'] and pkg['packages']`, thus the link would fail.

File: grapher/scan.py (hunk 2 above)
Comment (Unknwon) [inline grapher/scan.py]: Done.

File: grapher/scan.py (hunk 1 above)
Comment (beyang) [inline grapher/scan.py]: Ah, I see. What's a real-world example? Let's include a test case that covers this scenario.

Factor this out into a separate function with a docstring that includes the explanation you just gave above.

Comment (Unknwon) [inline grapher/scan.py]: Met this issue while testing on `httpie` (https://github.com/jkbrzt/httpie/blob/master/setup.py#L81), it only returns 4 packages, and not include `httpie.input` (it's a file) but in test file (https://github.com/jkbrzt/httpie/blob/master/tests/test_httpie.py#L4) it does import file as a module.

Comment (beyang) [inline grapher/scan.py]: Got it, makes sense. Can you factor this out into a separate function for readability and then add a test case repository (or modify one of the existing test case repositories to cover this)?

File: testdata/expected/python-link-tests/python-link-tests/PythonTestPackage.graph.json [hunk 3]
```diff
@@ -162,8 +195,18 @@
...
       "File": "pkg0/tests/test_m0.py",
-      "Start": 82,
-      "End": 85
+      "Start": 92,
+      "End": 95
+    },
+    {
+      "DefUnitType": "PythonTestPackage",
+      "DefUnit": "python-link-tests",
+      "DefPath": "test_m0.py/test_m0.bar.hello",
```
Comment (beyang) [inline testdata/expected/python-link-tests/python-link-tests/PythonTestPackage.graph.json]: Why does it not resolve this reference correctly? I would think that if it is able to resolve references to the module `m0`, it should also be able to resolve references to the module `m1`?

**Comment by Unknwon** [comment]
PR updated.

**Comment by beyang** [comment]
@Unknwon The structure of the python-link-tests repository now is abnormal. In particular, in `setup.py`, you pass `'app/pkg0'` to the `packages` arg of the `setup` function but "app/pkg0" is not the name of the package. The name of the package is "pkg0". Can you give an example of any other Python library that does it this way?

You also did not address the underlying issue of missing the `import m1` reference. Instead, you changed the test case to omit that reference. It's fine if we are explicitly not supporting that case now, but you need to given an explanation beyond "PR updated." Otherwise, the reviewer can't make sense of this change.

**Comment by Unknwon** [comment]
1. The `app/pkg0` issue has been fixed.
2. CI for `push` failed maybe because I use `git push --force` somehow breaks it
3. ... but the `pull request` one succeed and I re-run the task to make sure it pass as well.

**Comment by beyang** [comment]
LGTM, @Unknwon please update the Python srclib image and re-run coverage.
